tests/src/test/scala/com/stripe/agate/tensor/TensorTest.scala

```diff
@@ -824,4 +824,84 @@ object TensorTest extends Properties("TensorTest") {
 
     c1 && c2 && c3
   }
+
+  property("NonZero matches expected behavior") = {
+    val c0 = {
+      val t0 = tensor"[]"
+      val got = t0.nonZero
+      val expected = Tensor.const(DataType.Int64)(0, Shape.axes(1, 0))
+      Claim(got == expected)
+    }
+
+    val c1_1 = {
+      val t0 = tensor"[5, -1, 0, 4.4, 0, 666]"
+      val got = t0.nonZero
+      val expected = (tensor"[[0, 1, 3, 5]]").map(DataType.Int64)(_.toLong)
```

Comment:
Instead of map and toLong does cast

https://github.com/stripe/agate/blob/f03e1b131e397c0c0e45a22db7cf0233c16703d2/core/src/main/scala/com/stripe/agate/tensor/Tensor.scala#L183

Work?